src/tfrnnlmbin/lattice-lmrescore-tf-rnnlm-pruned.cc

```diff
@@ -0,0 +1,201 @@
+#include "util/common-utils.h"
+#include "lm/const-arpa-lm.h"
+#include "lat/kaldi-lattice.h"
+#include "lat/lattice-functions.h"
+#include "lat/compose-lattice-pruned.h"
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace kaldi::tf_rnnlm;
+    typedef kaldi::int32 int32;
+    typedef kaldi::int64 int64;
+    using fst::SymbolTable;
+    using fst::VectorFst;
+    using fst::StdArc;
+    using fst::ReadFstKaldi;
+
+    const char *usage =
+        "Rescores lattice with rnnlm that is trained with TensorFlow.\n"
+        "An example script for training and rescoring with the TensorFlow\n"
+        "RNNLM is at egs/ami/s5/local/tfrnnlm/run_lstm_fast.sh\n"
+        "\n"
+        "Usage: lattice-lmrescore-tf-rnnlm-pruned [options] [unk-file] \\\n"
+        "             <old-lm> <fst-wordlist> <rnnlm-wordlist> \\\n"
+        "             <rnnlm-rxfilename> <lattice-rspecifier> <lattice-wspecifier>\n"
+        " e.g.: lattice-lmrescore-tf-rnnlm-pruned --lm-scale=0.5 data/tensorflow_lstm/unkcounts.txt \\\n"
+        "              data/test/G.fst data/lang/words.txt data/tensorflow_lstm/rnnwords.txt \\\n"
+        "              data/tensorflow_lstm/rnnlm ark:in.lats ark:out.lats\n\n"
+        " e.g.: lattice-lmrescore-tf-rnnlm-pruned --lm-scale=0.5 data/tensorflow_lstm/unkcounts.txt \\\n"
+        "              data/test_fg/G.carpa data/lang/words.txt data/tensorflow_lstm/rnnwords.txt \\\n"
+        "              data/tensorflow_lstm/rnnlm ark:in.lats ark:out.lats\n";
+
+    ParseOptions po(usage);
+    int32 max_ngram_order = 3;
+    BaseFloat lm_scale = 0.5;
+    BaseFloat acoustic_scale = 0.1;
+    bool use_carpa = false;
+
+    po.Register("lm-scale", &lm_scale, "Scaling factor for <lm-to-add>; its negative "
+                "will be applied to <lm-to-subtract>.");
+    po.Register("acoustic-scale", &acoustic_scale, "Scaling factor for acoustic "
+                "probabilities (e.g. 0.1 for non-chain systems); important because "
+                "of its effect on pruning.");
+    po.Register("max-ngram-order", &max_ngram_order,
+        "If positive, allow RNNLM histories longer than this to be identified "
+        "with each other for rescoring purposes (an approximation that "
+        "saves time and reduces output lattice size).");
+    po.Register("use-const-arpa", &use_carpa, "If true, read the old-LM file "
+                "as a const-arpa file as opposed to an FST file");
+
+    KaldiTfRnnlmWrapperOpts opts;
+    ComposeLatticePrunedOptions compose_opts;
+    opts.Register(&po);
+    compose_opts.Register(&po);
+
+    po.Read(argc, argv);
+
+    if (po.NumArgs() != 7 && po.NumArgs() != 6) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    std::string lm_to_subtract_rxfilename, lats_rspecifier, rnn_word_list,
+      word_symbols_rxfilename, rnnlm_rxfilename, lats_wspecifier, unk_prob_file;
+    if (po.NumArgs() == 6) {
+      lm_to_subtract_rxfilename = po.GetArg(1),
```

Comment:
what does this do? lm_to_subtract_rxfilename? I think this the G.fst from the arpa model. Maybe call it decode_arpa_fst_rxfilename ? and change other names accordingly.

Reply:
i see your point but in general it might be the result of a previous rescoring pass, it might not be the one we decoded with.